evaluation/evaluate.py

```diff
@@ -194,81 +196,224 @@ def run_evaluation_from_config(working_dir: Path, config: dict, num_questions: i
+    theta = radar_factory(num_vars=len(metric_label_list))
+    fig, ax = plt.subplots(figsize=(6, 6), subplot_kw=dict(projection="radar"))
+    fig.subplots_adjust(wspace=0.5, hspace=0.20, top=0.85, bottom=0.05)
+
+    color = "c"  # Cyan
+
+    ax.set_title(title, weight="bold", size="large", horizontalalignment="center", verticalalignment="center", pad=20)
+    ax.plot(theta, data, color=color)
+    ax.fill(theta, data, facecolor=color, alpha=0.25, label="_nolegend_")
+    ax.tick_params(pad=15)
+    ax.set_rgrids([0, 1, 2, 3, 4, 5], angle=10)
+    ax.set_varlabels(metric_label_list)
+
+    plt.savefig(output_path)
+    plt.close(fig)
+
+
 def summarize_results_and_plot(
     questions_with_ratings: list, requested_metrics: list, results_dir: Path, passing_rate: int
 ):
     """Summarize the evaluation results and plot them."""
     df = pd.DataFrame(questions_with_ratings)
     summary = {}
-    metric_list, metric_name = [], []
-    pass_rate, mean_rate = [], []
-    min_list, mean_list, max_list = [], [], []
+    gpt_metric_list, gpt_metric_mean_list, numeric_metric_list = [], [], []
+    rating_stat_data = {
+        "pass_count": {},
+        "pass_rate": {},
+        "mean_rating": {},
+    }
+    num_stat_data = {"latency": {}, "f1_score": {}, "answer_length": {}}
+    aggregated_data_lists = {}
+
+    width = 0.4  # the width of the bars
+
+    gpt_metrics = ["gpt_groundedness", "gpt_relevance", "gpt_coherence", "gpt_similarity", "gpt_fluency"]
+
+    numeric_metrics = ["latency", "f1_score", "answer_length"]
+
+    display_metric_name = {
+        "pass_count": "Pass Count",
+        "pass_rate": "Pass Rate",
+        "mean_rating": "Average Rating",
+        "latency": "Latency (sec)",
+        "f1_score": "F1 Score",
+        "answer_length": "Answer Length (words)",
+        "gpt_groundedness": "GPT Groundedness Rating",
+        "gpt_relevance": "GPT Relevance Rating",
+        "gpt_coherence": "GPT Coherence Rating",
+        "gpt_similarity": "GPT Similarity Rating",
+        "gpt_fluency": "GPT Fluency Rating",
+    }
+
+    short_metric_name = {
+        "pass_count": "Pass Count",
+        "pass_rate": "Pass Rate",
+        "mean_rating": "Avg. Rating",
+        "latency": "Latency",
+        "f1_score": "F1 Score",
+        "answer_length": "Length",
+        "gpt_groundedness": "Groundness",
+        "gpt_relevance": "Relevance",
+        "gpt_coherence": "Coherence",
+        "gpt_similarity": "Similarity",
+        "gpt_fluency": "Fluency",
+    }
+
+    metric_value_labels = {
+        "pass_count": "Number of Questions",
+        "pass_rate": "Percentage",
+        "mean_rating": "Rating Score",
+        "latency": "Seconds",
+        "f1_score": "F1 Score",
+        "answer_length": "Words",
+        "gpt_groundedness": "Rating Score",
+        "gpt_relevance": "Rating Score",
+        "gpt_coherence": "Rating Score",
+        "gpt_similarity": "Rating Score",
+        "gpt_fluency": "Rating Score",
+    }
```

Comment:
Does it make sense to hardcode stuff like that here? What if we introduce new metrics?
I think it'd be better if we introduced the display/short names and labels as class attributes of our metrics (both custom and built-in ones).

Reply:
surely using a constant dict to state their value isn't the best approach I can tell, and apparently making them class attributes is a better way. I was just thinking is that really worth it to change them, would anyone apart from us really going to add more metrics? (btw "pass_count", "pass_rate" and "mean_rating" still need to be separately handled)

Reply:
> would anyone apart from us really going to add more metrics?

@GrayNekoBean haha that's highly unlikely for this project, but still, maintaining these directly at their source is way more scalable. I would also argue that the short name is probably redundant.